lib/publishing.gradle

```diff
@@ -0,0 +1,45 @@
+import java.time.ZonedDateTime
+
+// Download and extract the Cloud Profiler Java Agent
+ext {
+    // where to place the Cloud Profiler agent in the container
+    cloudProfilerLocation = "/opt/cprof"
+
+    // location for jib extras, including the Java agent
+    jibExtraDirectory = "${buildDir}/jib-agents"
+}
+task downloadProfilerAgent(type: Download) {
+    // where to download the Cloud Profiler agent https://cloud.google.com/profiler/docs/profiling-java
+    src "https://storage.googleapis.com/cloud-profiler/java/latest/profiler_java_agent.tar.gz"
+    dest "${buildDir}/cprof_java_agent_gce.tar.gz"
+}
+task extractProfilerAgent(dependsOn: downloadProfilerAgent, type: Copy) {
+    from tarTree(downloadProfilerAgent.dest)
+    into "${jibExtraDirectory}/${cloudProfilerLocation}"
+}
+
+jib {
```

Comment:
This is a library, not an app or service, so it shouldn't need `jib` at all. You may want to think about how you're going to package up the library as a separate entity from the CLI if you plan to do that.